src/Sarif.Multitool/Program.cs

```diff
@@ -12,33 +15,42 @@ internal static class Program
-                (TransformOptions transformOptions) => new TransformCommand().Run(transformOptions),
-                (MergeOptions mergeOptions) => new MergeCommand().Run(mergeOptions),
-                (RebaseUriOptions rebaseOptions) => new RebaseUriCommand().Run(rebaseOptions),
-                (AbsoluteUriOptions absoluteUriOptions) => new AbsoluteUriCommand().Run(absoluteUriOptions),
-                (PageOptions pageOptions) => new PageCommand().Run(pageOptions),
-                (QueryOptions queryOptions) => new QueryCommand().Run(queryOptions),
-                (ResultMatchingOptions baselineOptions) => new ResultMatchingCommand().Run(baselineOptions),
-                (ResultMatchSetOptions options) => new ResultMatchSetCommand().Run(options),
-                (FileWorkItemsOptions fileWorkItemsOptions) => new FileWorkItemsCommand().Run(fileWorkItemsOptions),
-                errs => CommandBase.FAILURE);
+            try
+            {
+                return Parser.Default.ParseArguments<
+                    ValidateOptions,
+                    ConvertOptions,
+                    RewriteOptions,
+                    TransformOptions,
+                    MergeOptions,
+                    RebaseUriOptions,
+                    AbsoluteUriOptions,
+                    PageOptions,
+                    QueryOptions,
+                    ResultMatchingOptions,
+                    ResultMatchSetOptions,
+                    FileWorkItemsOptions>(args)
+                  .MapResult(
+                    (ValidateOptions validateOptions) => new ValidateCommand().Run(validateOptions),
+                    (ConvertOptions convertOptions) => new ConvertCommand().Run(convertOptions),
+                    (RewriteOptions rewriteOptions) => new RewriteCommand().Run(rewriteOptions),
+                    (TransformOptions transformOptions) => new TransformCommand().Run(transformOptions),
+                    (MergeOptions mergeOptions) => new MergeCommand().Run(mergeOptions),
+                    (RebaseUriOptions rebaseOptions) => new RebaseUriCommand().Run(rebaseOptions),
+                    (AbsoluteUriOptions absoluteUriOptions) => new AbsoluteUriCommand().Run(absoluteUriOptions),
+                    (PageOptions pageOptions) => new PageCommand().Run(pageOptions),
+                    (QueryOptions queryOptions) => new QueryCommand().Run(queryOptions),
+                    (ResultMatchingOptions baselineOptions) => new ResultMatchingCommand().Run(baselineOptions),
+                    (ResultMatchSetOptions options) => new ResultMatchSetCommand().Run(options),
+                    (FileWorkItemsOptions fileWorkItemsOptions) => new FileWorkItemsCommand().Run(fileWorkItemsOptions),
+                    errs => CommandBase.FAILURE);
+            }
+            finally 
+            {
+                ITelemetryChannel channel = ServiceProviderFactory.ServiceProvider.GetService<ITelemetryChannel>();
+                channel?.Flush();
```

Comment:
>Flush [](start = 25, length = 5)

@michaelcfanning, we have to flush the telemetry stream before exiting. Are you OK doing it here? Or do you want to move this to a try/finally in the SarifWorkItemFiler?

Reply:
It feels like the SarifWorkItemFiler would be the better place, simply because we know we often won't be used from this specific command line entry.  I'd prefer to keep everything crucial within the workitemfiler, even if in some cases it might not be necessary (when we run as a service, for example).

Reply:
Fixed. I moved the flush to the Dispose method of SarifWorkItemFiler.

---
In reply to: [396154484](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r396154484) [](ancestors = 396154484)